store/src/java/com/zimbra/cs/stats/ZimbraPerf.java

```diff
@@ -432,6 +432,14 @@ public static void incrementPrepareCount() {
      * during realtime stats collection.
      */
     public static void addStatsCallback(RealtimeStatsCallback callback) {
+        if (realtimeStats == null) {
+            ZimbraLog.perf.debug("Call to addStatsCallback when realtimeStats has not been initialized\n%s",
+                    ZimbraLog.getStackTrace(15));
+            /* This probably happens inside a commandline tool like zmmetadump where it doesn't
+             * make sense to mix in stats with those of the main Zimbra process.
+             */
+            return;
+        }
         realtimeStats.addCallback(callback);
```

Comment:
Is this the location of the NPE?

Reply:
Yes.  The assumption of the new code is that this method has been called in error and can be safely ignored.  Could have tried to fix the root cause but quite likely that new code would trip up again as people findling with DB Pool initialization (in this case) may not think of all the use cases.